stdlib/src/ssl.rs

```diff
@@ -174,90 +175,75 @@ mod _ssl {
         parse_version_info(openssl_api_version)
     }
 
+    macro_rules! ssl_exception_wrapper {
+        ($exc_name:literal, {$($base:expr), +}, $meta_class:expr) => {{
+            rustpython_common::static_cell! {
+                static ERROR: PyTypeRef;
+            }
+            ERROR
+                .get_or_init(|| {
+                    PyType::new_ref(
+                        concat!("ssl.", $exc_name),
+                        vec![$($base), +],
+                        Default::default(),
+                        PyBaseException::make_slots(),
```

Comment:
Although original `new_ref` for `SSLError` use `PyBaseException::make_slots()` and `new_simple_ref` for the other errors use `Default::default()`, I replaced them as just single `PyBaseException::make_slots()` as I think there are no difference between them in cpython.

Reply:
I'm thinking this should apply to more errors, not just `ssl` (i.e the places where `PyErr_NewException` is used). Using the same macro for errors in the other modules #3397 didn't work?

Reply:
Hmm.. I got it though, it seems a lot complex than I think. 
It seems `zlib.error` can be replaced by current macro, though `_csv` use `PyType_FromModuleAndSpec` with `PyExc_Exception` as bases arguments. I dont know what is the difference between them yet..

Reply:
And current module-level exceptions have wrong `__module__` as its attribute
#### rustpython
```python
>>>>> csv.Error.__module__
>>>>> SSLCertVerificationError.__module__
'builtins'
>>>>> zlib.error.__module__
'builtins'
```

#### cpython
```python
>>> csv.Error.__module__
'_csv'
>>> _ssl.SSLCertVerificationError.__module__
'ssl'
>>> zlib.error.__module__
'zlib'
```

Reply:
I think there need to `PyErr_NewException` which create exception with given bases and also parse module name from given exception separated by dot('.').

Reply:
What is current `vm.new_exception` exactly for? I wonder if it is somewhat originally intended to use in there cases,.

Reply:
I think #3396 might be what we need here, when that lands the macro could just invoke `vm.new_class` it seems.

Reply:
Oh. It seems so 😊. After #3396 merged, I'll modify the macro to use it and apply it to other errors.

Reply:
Merged!

Reply:
Ok, so, I don't think this needs a macro tbh. It would probably be better to just create a new function (in ctx), name it `new_exception_type` and duplicate what `new_class` is doing. 

Looking (quite back!) into why `new_class` was added, it appears it was meant as a way to create exceptions ([ref](https://github.com/RustPython/RustPython/commit/f76472c00b7fc32ec2bf821539e3b321c3bfd3aa)). I think it is misnamed and currently confusing.

`vm.new_exception` -> Create a new exception instance.
`vm.ctx.new_exception_type` -> Create a new exception (specifically for module exceptions, mirroring what `PyErr_NewException` does).

It should definitely take a name, a module, and an optional docstring (`_ssl` errors seem to have docs attached to them). It should then forward everything to `PyType::new_ref`). I *think* this way it would make it usable for all module exceptions we currently need to create.

Reply:
Sorry for being late! Thanks for the detailed advice!  I was meant to modify macro itself to something other (sorry for bad English 😂). but your idea seems much better than what I originally planned to do. 

I would ping you after implementation (I have no time to work on it today, sorry .😢)

Reply:
@DimitrisJim Long time no see! I add some procedural macro code for wrapping each function which have `#[pyattr]` in [b761e69](https://github.com/RustPython/RustPython/pull/3402/commits/b761e6991d3ac6f12005aa4cf15912deae5d8f5f) and [ae9f85c](https://github.com/RustPython/RustPython/pull/3402/commits/ae9f85c2b9bcc3dd2627d9a32c3bb3bcc3050f2c).

We planned to pass docstring to `new_exception_type` like cpython, though I think automatic document parsing with comment prefix with `///` would be better. 

But it is currently not working as we expected. This maybe handled in separate PR.

Reply:
> I add some procedural macro code for wrapping each function which have #[pyattr]

that's an interesting approach! It should also work fine for non-error `#[pyattr] fn`'s I'm supposing.

> We planned to pass docstring to new_exception_type like cpython, though I think automatic document parsing with comment prefix with /// in rustpython would be better.

Right, I don't remember if we actually try and grab docstrings for `pyattr`s, but that is definitely a separate PR.